GitUpKit/Interface/GIGraphView.m

```diff
@@ -860,97 +860,168 @@ static void _DrawLine(GILine* line, CGContextRef context, CGFloat offset, CGFloa
+    NSString* branchName = remoteBranch.branchName;
+    [multilineTitle appendFormat:@"%@/%@\n", remoteName, branchName];
+    NSRange branchNameRange = NSMakeRange(multilineTitle.length - branchName.length - 1, branchName.length);
+    [boldRanges addObject:[NSValue valueWithRange:branchNameRange]];
+    [darkRanges addObject:boldRanges.lastObject];
+  }
+  
+  if (branch.remoteBranches.count > 0) {
+    [multilineTitle appendString:@"\n"];
   }
-  index = 0;
+  
   for (GCHistoryTag* tag in branch.tags) {
-    if (index) {
-      [label appendString:@", "];
-    } else {
-      if (label.length) {
-        [label appendString:@"\n"];
-      }
-      [label appendString:@"["];
-    }
-    [label appendString:tag.name];
-    ++index;
+    NSString* tagName = tag.name;
+    [multilineTitle appendFormat:@"[%@]\n", tagName];
+    NSRange tagNameRange = NSMakeRange(multilineTitle.length - tagName.length - 2, tagName.length);
+    [boldRanges addObject:[NSValue valueWithRange:tagNameRange]];
+    [darkRanges addObject:boldRanges.lastObject];
   }
-  if (index) {
-    [label appendString:@"]"];
+  
+  if (branch.tags.count > 0) {
+    [multilineTitle appendString:@"\n"];
   }
-  if (!label.length) {
-    [label release];
-    return;  // This should only happen if we have a detached HEAD with no other references pointing to the commit
+  
+  if (multilineTitle.length == 0) {
+    [boldRanges release];
+    [darkRanges release];
+    [multilineTitle release];
+    return; // This should only happen if we have a detached HEAD with no other references pointing to the commit
   }
   
-  // Prepare text
+  // Create a light string to show above the HEAD
+  
+  CTFontRef titleFont = CTFontCreateUIFontForLanguage(kCTFontUIFontSystem, 12.0, CFSTR("en-US"));
+  NSMutableParagraphStyle *style = [[NSMutableParagraphStyle alloc] init];
+  style.lineHeightMultiple = 0.8;
+  NSDictionary* multilineTitleAttributes = @{ NSFontAttributeName: (id)titleFont, NSForegroundColorAttributeName: color, NSParagraphStyleAttributeName: style };
+  NSMutableAttributedString* multilineAttributedTitle = [[NSMutableAttributedString alloc] initWithString:multilineTitle attributes:multilineTitleAttributes];
+  [style release];
+  [multilineTitle release];
+  
+  // Change font to bold on ranges collected before
+  
+  NSFont* boldFont = (NSFont *)CTFontCreateUIFontForLanguage(kCTFontUIFontEmphasizedSystem, 12.0, CFSTR("en-US"));
+  for (NSValue* bold in boldRanges) {
+    [multilineAttributedTitle addAttribute:NSFontAttributeName value:boldFont range:bold.rangeValue];
+  }
+  [boldFont release];
+  [boldRanges release];
+  
+  // Change color to dark on ranges collected before
+  
+  NSColor* darkColor = [color shadowWithLevel:0.8];
+  for (NSValue* dark in darkRanges) {
+    [multilineAttributedTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:dark.rangeValue];
+  }
+  [darkRanges release];
+  
+  // Create CoreFoundation string from Foundation
```

Comment:
You can remove this comment and avoid the copy: just pass `multilineAttributedTitle` wherever a `CFAttributedStringRef` is expected.